homeassistant/components/madvr/sensor.py

```diff
@@ -0,0 +1,239 @@
+        name="GPU Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        value_fn=lambda coordinator: get_temperature(coordinator, "temp_gpu"),
+        translation_key="temp_gpu",
+    ),
+    MadvrSensorEntityDescription(
+        key="temp_hdmi",
+        name="HDMI Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        value_fn=lambda coordinator: get_temperature(coordinator, "temp_hdmi"),
+        translation_key="temp_hdmi",
+    ),
+    MadvrSensorEntityDescription(
+        key="temp_cpu",
+        name="CPU Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        value_fn=lambda coordinator: get_temperature(coordinator, "temp_cpu"),
+        translation_key="temp_cpu",
+    ),
+    MadvrSensorEntityDescription(
+        key="temp_mainboard",
+        name="Mainboard Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        value_fn=lambda coordinator: get_temperature(coordinator, "temp_mainboard"),
+        translation_key="temp_mainboard",
+    ),
+    MadvrSensorEntityDescription(
+        key="incoming_res",
+        name="Incoming Resolution",
+        value_fn=lambda coordinator: coordinator.data.get("incoming_res"),
+        translation_key="incoming_res",
+    ),
+    MadvrSensorEntityDescription(
+        key="incoming_frame_rate",
+        name="Incoming Frame Rate",
+        value_fn=lambda coordinator: coordinator.data.get("incoming_frame_rate"),
+        translation_key="incoming_frame_rate",
+    ),
+    MadvrSensorEntityDescription(
+        key="incoming_color_space",
+        name="Incoming Color Space",
+        value_fn=lambda coordinator: coordinator.data.get("incoming_color_space"),
+        translation_key="incoming_color_space",
+    ),
+    MadvrSensorEntityDescription(
+        key="incoming_bit_depth",
+        name="Incoming Bit Depth",
+        value_fn=lambda coordinator: coordinator.data.get("incoming_bit_depth"),
+        translation_key="incoming_bit_depth",
+    ),
+    MadvrSensorEntityDescription(
+        key="incoming_colorimetry",
+        name="Incoming Colorimetry",
+        value_fn=lambda coordinator: coordinator.data.get("incoming_colorimetry"),
+        translation_key="incoming_colorimetry",
+    ),
+    MadvrSensorEntityDescription(
+        key="incoming_black_levels",
+        name="Incoming Black Levels",
+        value_fn=lambda coordinator: coordinator.data.get("incoming_black_levels"),
+        translation_key="incoming_black_levels",
+    ),
```

Comment:
What are the possible values

Reply:
in order
1. 422, 444, 420, RGB
2. 10bit, 12bit, 8bit
3. 601|PAL|709|DCI|2020
4. PC | TV

Reply:
If these are all pre known we should use enum sensors to allow users to be able to preselect them in the automation editor and know what the possible values are